### DISCUSSION TITLE
Don’t request $schema url during validation

### DISCUSSION THREAD

**Discussion by SylvieManx** [networknt/json-schema-validator#1074] [Q&A] [answered]
How can I set JsonSchema to not read the $schema attribute in json?
I don't want to request this $schema during validation, but I don't see how to set this parameter in the source code
 ERROR c.networknt.schema.JsonSchemaFactory - Failed to load json schema from https://schemas.amazon.com/selling-partners/definitions/product-types/meta-schema/v1
![json](https://github.com/networknt/json-schema-validator/assets/20614254/9d11377b-5479-460d-8fbf-fe89f1759e36)

**Comment by justin-tay** [comment, marked as answer]
If you want to use a custom meta schema you need to tell the library what it is or it will attempt to load it to determine what vocabularies are defined in the meta schema.

```java
JsonMetaSchema metaSchema = JsonMetaSchema
        .builder("https://schemas.amazon.com/selling-partners/definitions/product-types/meta-schema/v1",
                JsonMetaSchema.getV201909())
        .keyword(new MaxUniqueItemsKeyword())
        .keyword(new MaxUtf8ByteLengthKeyword())
        .keyword(new MinUtf8ByteLengthKeyword()).build();
JsonSchemaFactory factory = JsonSchemaFactory.builder().defaultMetaSchemaIri(metaSchema.getIri())
        .metaSchema(metaSchema).build();
```

**Reply by SylvieManx** [reply]
Thank you very much for your professional answer, it perfectly solved my problem
